rewrite-core/src/main/java/org/openrewrite/ExcludeFileFromGitignore.java

```diff
@@ -127,17 +127,18 @@ private List<String> sortRules(String[] originalRules, List<String> newRules) {
                 return distinctValuesStartingReversed(results);
             }
 
-            private <T> List<T> distinctValuesStartingReversed(List<T> list) {
-                Set<T> set = new LinkedHashSet<>();
-                ListIterator<T> iterator = list.listIterator(list.size());
+            private List<String> distinctValuesStartingReversed(List<String> list) {
+                LinkedList<String> filteredList = new LinkedList<>();
```

Comment:
Is there any benefit to using LinkedList here over a regular ArrayList? Or can we move to ArrayList instead?
```suggestion
                List<String> filteredList = new ArrayList<>();
```

Reply:
I believe there is. 

`addFirst` is the main thing I believe. 
1. It’s easier to read VS adding on index 0
2. When inserting I believe internally not all objects need to be shifted 1 index. Might be wrong on that benefit ☺️